### PR TITLE
Fix a simple problem

### PR DESCRIPTION
**Pull request by imkiva**
bors merge

### CODE REVIEW

**Comment by imkiva** [comment]
bors cancel

**Comment by imkiva** [comment]
bors merge

File: lsp/src/main/java/org/aya/lsp/utils/Resolver.java [hunk 1]
```diff
@@ -156,7 +157,9 @@
     }
 
     @Override public void visitSignatured(@NotNull Signatured signatured, XY xy) {
-      signatured.telescope.forEach(tele -> check(xy, tele.ref(), tele.sourcePos()));
+      signatured.telescope
+        .filterNot(tele -> tele.ref().name().startsWith(Constants.ANONYMOUS_PREFIX))
```
Comment (ice1000) [inline lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: What if this name is written by users?

Comment (imkiva) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: We should reserve the meaning of names starting with an underscore. It means _does not matter_. If users any term uses those names the resolver should raise an warning or error.

Comment (ice1000) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: Is this just a performance improvement? What if there's a lambda expression in an anonymous parameter?

Comment (imkiva) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: > Is this just a performance improvement?

Yes, finding usages on each LSP action (like hover) is very slow.

> What if there's a lambda expression in an anonymous parameter?

I don't get it. Care to give an example?

Comment (ice1000) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: > We should reserve the meaning of names starting with an underscore. It means _does not matter_. If any term uses those names the resolver should raise a warning or even an error.

I agree

Comment (ice1000) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: > I don't get it. Care to give an example?

Like `def test (Ok (\ x => x))`, the above code implies that this `Ok (\ x => x)` will not be searched in this visitor. Is this intended?

Comment (imkiva) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: > > I don't get it. Care to give an example?
> 
> Like `def test (Ok (\ x => x))`, the above code implies that this `Ok (\ x => x)` will not be searched in this visitor. Is this intended?

No, the type is always searched

Comment (imkiva) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: This PR fixes searching `PsiNamedElement` alike things, that is an anonymous parameter never introduces a usable name.

Comment (ice1000) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: Oh I see, it's searched in `super.visitSignatured`, tyvm!

Comment (ice1000) [reply lsp/src/main/java/org/aya/lsp/utils/Resolver.java:161]: bors r+

**Pull request merged**